Assets/PyroParticles/Prefab/Script/MeteorSwarmScript.cs

```diff
@@ -74,7 +74,7 @@ private IEnumerator SpawnMeteor()
             }
 
             // find a random source and destination point within the specified radius
-            Vector3 src = Source + (UnityEngine.Random.insideUnitSphere * SourceRadius);
+            Vector3 src = transform.position + new Vector3(0, SpawnHeight, 0) + (UnityEngine.Random.insideUnitSphere * SourceRadius);
```

Comment:
I might be wrong, but I think that `insideUnitSphere` may cause a downward vector. Which would cause the meteor to spawn closer to the ground, or even under the ground depending on what SourceRadius and SpawnHeight are.

Reply:
As long as SpawnHeight is reasonably high, this won't be a problem.